gui

```diff
@@ -111,67 +111,72 @@ guimode="$(cat "${DIRECTORY}/data/settings/App List Style")"
-    # workaround is to get the first output returned for the connected display
-    screen_dimensions="$(xrandr --nograb --current | awk -F 'connected |\\+|\\('  '/ connected.*[0-9]+x[0-9]+\+/ && $2 {printf $2 ", "}' | tr 'x+' ' ' | tr ',+' ' ')"
+  if [ -z "$screen_width" ] || [ -z "$screen_height" ];then
+    screen_dimensions="$(xrandr --nograb --current | awk -F 'connected |\\+|\\('  '/ connected.*[0-9]+x[0-9]+\+/ && $2 {printf $2 ", "}' | sed -n -e 's/^.*primary //p' | tr 'x+' ' ' | tr ',+' ' ')"
+    if [ -z "$screen_dimensions" ];then
+      # if screen_dimensions is empty, this could be a single monitor wayland display which does not have the word "primary" in the output
+      # workaround is to get the first output returned for the connected display
+      screen_dimensions="$(xrandr --nograb --current | awk -F 'connected |\\+|\\('  '/ connected.*[0-9]+x[0-9]+\+/ && $2 {printf $2 ", "}' | tr 'x+' ' ' | tr ',+' ' ')"
+    fi
+    screen_width="$(awk '{print $1}' <<<"$screen_dimensions")"
+    screen_height="$(awk '{print $2}' <<<"$screen_dimensions")"
   fi
-  screen_width="$(awk '{print $1}' <<<"$screen_dimensions")"
-  screen_height="$(awk '{print $2}' <<<"$screen_dimensions")"
-  unset screen_dimensions
   
   if [[ "$guimode" == yad* ]];then
-
+    
     # if this is being run by yad, there should be a yad window id
     # store in arrary of x and y
     # this allows for windows to be moved by the user and the second window to still show up where expected
     main_yad_window=$YAD_XID
     if [ -n "${main_yad_window}" ]; then
       window_info=$(xwininfo -id $main_yad_window)
-      # for speed of execution, we are assuming the order of matches from xwininfo
-      # this is safe as the output order has been standardized for over a decade
-      # this is the location of where the window content starts inside the title bar and window borders (pos x then pos y)
-      main_yad_window_pos=($(echo "$window_info" | sed -n 's/Absolute upper-left.*://p'))
-      # these are the dimensions of where the window content starts inside the title bar and window borders (dim x then dim y)
-      main_yad_window_dim=($(echo "$window_info" | sed -n 's/Width://p;s/Height://p'))
-      # again, for speed, all border widths are saved to one array
-      # left, right, title, bottom [0, 1, 2, 3]
-      # this is the size of the window borders (including title bar)
-      border_widths=($(xprop _NET_FRAME_EXTENTS -id "$main_yad_window" | sed -n  's/_NET_FRAME_EXTENTS.*=//p' | tr ',' '\n'))
-      #total dimensions for both yad windows side by side
-      if [ $screen_width -le 1000 ] || [ $screen_height -le 600 ];then
-        #gui size for small screens
-        height=${main_yad_window_dim[1]}
-        width=600
+      
+      #skip calculating offsets if yad window is in the same place as last time
+      if [ -z "$last_window_info" ] || [ -z "$geometry1" ] || [ -z "$geometry2" ] || [ "$last_window_info" != "$window_info" ];then
```

Comment:
same issue as above. best to just drop this commit unless there was actually some functional issues that were fixed.

Reply:
I made these changes to get_positions to see how much of a difference it would make when rapidly selecting between apps, and the difference was more noticeable when there was high load on the system, like a program being compiled, a video playing, or a sudo apt upgrade.
Otherwise the difference is not really noticeable.

Reply:
can't find any issue with it